File: solutions/Leetcode_775/leetcode_775.cc
#include <cmath>
#include <iostream>
#include <vector>

using namespace std;

class Solution1
{
public:
    bool isIdealPermutation(vector<int> &A)
    {
        for (int i = 0; i != A.size(); ++i)
            if (abs(A[i] - i) > 1)
                return false;
        return true;
    }
};

int main()
{
    vector<int> A{1, 0};
    bool ans = Solution1().isIdealPermutation(A);
    cout << ans << endl;
}

File: solutions/Leetcode_775/leetcode_775.py
""" Leetcode 775 - Global and Local Inversions

https://leetcode.com/problems/global-and-local-inversions/

1. Time: O(n) Memory: O(1)

"""

from typing import List


class Solution1:
    """ 1. MINE |  SPECIFIC """

    def is_ideal_permutation(self, A: List[int]) -> bool:
        for i in range(len(A)):
            if abs(A[i] - i) > 1:
                return False
        return True


if __name__ == '__main__':
    A = [1, 2, 0]
    ans = Solution1().is_ideal_permutation(A)
    print(ans)
